modelon/impact/client/client.py

```diff
@@ -245,10 +293,81 @@ def upload_workspace(self, path_to_workspace):
             resp["id"],
             self._sal.workspace,
             self._sal.model_executable,
             self._sal.experiment,
             self._sal.custom_function,
         )
+
+    def get_projects(self, vcs_info: bool = True):
+        # TODO: Return a list of project entities
+        return self._sal.project.projects_get(vcs_info=vcs_info)
+
+    def _get_versioned_project_meta_from_ids(
+        self, vcs_project_ids: List[str]
+    ) -> List[VCSProjectMeta]:
+        projects = self.get_projects(vcs_info=True)["data"]["items"]
+        return [
+            VCSProjectMeta(project['id'], project['definition']['name'])
+            for project in projects
+            if project['id'] in vcs_project_ids
+        ]
+
+    def _get_resolved_vcs_matchings(self, shared_definition: WorkspaceDefinition):
+        matchings = self.get_vcs_matchings(
+            shared_definition, only_multiple_matches=True
+        )
+        entries = []
+        for matching in matchings:
+            entries += matching.interactive_resolve()
+        return entries
+
+    def _import_from_shared_definition(self, shared_definition: WorkspaceDefinition):
+        resp = self._sal.workspace.import_from_shared_definition(
+            shared_definition.to_dict()
+        )
+        operation = WorkspaceImportOperation(
+            resp["data"]["location"],
+            self._sal.workspace,
+            self._sal.model_executable,
+            self._sal.experiment,
+            self._sal.custom_function,
+        )
+        return operation
+
+    def import_from_shared_definition(
+        self, shared_definition: WorkspaceDefinition, interactive: bool = True,
+    ):
+        operation = self._import_from_shared_definition(shared_definition)
+        if interactive:
+            try:
+                return operation.wait()
+            except exceptions.IllegalWorkspaceImport as exe:
+                entries = self._get_resolved_vcs_matchings(shared_definition)
+                if entries:
+                    operation = self._import_from_shared_definition(
+                        shared_definition.with_selected_matchings(entries)
+                    )
+                    return operation.wait()
+                raise exe
+        return operation
+
+    def get_vcs_matchings(
+        self, shared_definition: WorkspaceDefinition, only_multiple_matches: bool = True
+    ) -> List[VCSMatching]:
+        vcs_matchings = []
+        matchings = self._sal.workspace.get_vcs_matchings(shared_definition.to_dict())[
+            'data'
+        ]['vcs']
+        for entry in matchings:
+            projects = entry['projects']
+            if only_multiple_matches and len(projects) == 1:
+                continue
+            vcs_uri = shared_definition.get_vcs_uri_from_reference_id(entry["entryId"])
+            projects = self._get_versioned_project_meta_from_ids(
+                vcs_project_ids=[project['id'] for project in projects]
+            )
```

Comment:
If this is needed then I think we should change the return from 'matchings' to contain all the project data instead of trying to get it again.